### PR TITLE
Global update (2.55, CLEAR button, port, launch/attach for win32)

### PR DESCRIPTION
**Pull request by kenny94xxhell**
Is it cleaner now ?

### CODE REVIEW

File: labot/mitm/bridge.py [hunk 1]
```diff
@@ -215,5 +224,10 @@ def handle(self, data, origin):
                 self.dumper.dump(msg)
             self.other[origin].sendall(msg.bytes())
 
+            self.handle_message(parsedMsg, origin)
             msg = Msg.fromRaw(self.buf[origin], from_client)
 
+            time.sleep(0.005)
+
+    def handle_message(self, m, o):
+        pass
```
Comment (louisabraham) [inline labot/mitm/bridge.py:233]: good idea!

File: labot/mitm/bridge.py [hunk 2]
```diff
@@ -215,5 +224,10 @@ def handle(self, data, origin):
                 self.dumper.dump(msg)
             self.other[origin].sendall(msg.bytes())
 
+            self.handle_message(parsedMsg, origin)
             msg = Msg.fromRaw(self.buf[origin], from_client)
 
+            time.sleep(0.005)
```
Comment (louisabraham) [inline labot/mitm/bridge.py:230]: very nice

File: scripts/mitm.py [hunk 3]
```diff
@@ -15,11 +16,13 @@
 from fritm import hook, start_proxy_server
 import frida
 
+appdata = os.getenv("appdata")
+parent = os.path.dirname(appdata)
```
Comment (louisabraham) [inline scripts/mitm.py]: appdata is None on linux (probably same on macOS too)

File: scripts/mitm.py [hunk 4]
```diff
@@ -95,7 +94,10 @@ def my_callback(coJeu, coSer):
         target = dofus.pid
 
     if args.attach:
-        target = args.pid or "dofus"
+        if sys.platform == "linux" or sys.platform == "win32":
```
Comment (louisabraham) [inline scripts/mitm.py]: I dont think it's "Dofus.exe" on linux

**Comment by louisabraham** [comment]
Merciii :smiling_face_with_three_hearts:

File: scripts/mitm.py (hunk 4 above)
Comment (kenny94xxhell) [reply scripts/mitm.py]: Yes it is. On linux, Dofus.exe is executed by Wine

**Pull request merged**